lib/archethic/self_repair/network_view.ex

```diff
@@ -0,0 +1,203 @@
+defmodule Archethic.SelfRepair.NetworkView do
+  @moduledoc """
+  The network view is 2 things:
+  - the P2P view (list of available & authorized nodes)
+  - the Network chains view (oracle/origin/nodesharedsecrets)
+
+  It is useful to compare with other nodes to detect desynchronization.
+  The P2P view is handled by the P2P module, we just do the hash here for convenience.
+  """
+
+  use GenServer
+  @vsn Mix.Project.config()[:version]
+
+  alias Archethic.OracleChain
+  alias Archethic.Crypto
+  alias Archethic.P2P
+  alias Archethic.PubSub
+  alias Archethic.SharedSecrets
+  alias Archethic.TransactionChain
+  alias Archethic.TransactionChain.Transaction
+
+  require Logger
+
+  defmodule State do
+    defstruct [
+      # binary()
+      node_shared_secrets: nil,
+      # binary()
+      oracle: nil,
+      # list(binary())
+      origin: [],
+      chains_hash: <<>>
+    ]
+  end
+
+  # ------------------------------------------------------
+  #               _
+  #    __ _ _ __ (_)
+  #   / _` | '_ \| |
+  #  | (_| | |_) | |
+  #   \__,_| .__/|_|
+  #        |_|
+  # ------------------------------------------------------
+
+  @doc """
+  Start the NetworkView server
+  """
+  @spec start_link(list()) :: GenServer.on_start()
+  def start_link([]) do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  @doc """
+  Return the hash of the P2P view
+  """
+  @spec get_p2p_hash() :: binary()
+  def get_p2p_hash() do
+    # this is not using the genserver
+    # since the nodes' state is already in the P2P module
+    P2P.authorized_and_available_nodes()
```

Comment:
I think this can also be in the state of the genserver, but just the hash. Because actually every time a new transaction is received or sent, you will compute the hash which will almost be the same all the time. This is a useless effort I think.
You can also use the `load_transaction` function on the `node` transaction and update the hash when you receive a new one.
Then the hash will stay in the state and the `get_p2p_hash` will just do a call to the genserver to get the state